### ISSUE TITLE
Metric comparison - Colab code - Validation set missing

### ISSUE THREAD

**Issue by simoneroviaro** [mrdbourke/zero-to-mastery-ml#111]
The issue refer to the follow-up Google Colab code of the "Note: Metric Comparison Improvement", by the end of the chapter "Scikit-learn: Creating Machine Learning Models".

In the Colab code, both RandomizedSearchCV and GridSearchCV were applied directly to the training set without an explicit validation set. 

Quote "The most important part is they all use the same data splits created using train_test_split() and np.random.seed(42)".

I initially supposed that this was referring to the fact that during the previous lessons, a validation set was created for RandomizedSearchCV, but it was not consistent with the GridSearchCV, where a 80/20 train_test_split was used instead.

This turned out not to be the case in Colab code. Infact both RandomizedSearchCV and GridSearchCV were applied directly to the training set without an explicit validation set. 

- Couldn't this approach lead to overfitting? Any tuning process based on test set performance indirectly leaks information about the test set into the model selection process. 

This is not consistent with the content of the previous lessons where the validation set was exaplained.

Could you please clarify?
Thanks,
Simone.

**Comment by 0quaaD** [comment]
Hey!
You are correct that using the test set for tuning can result in information leakage and overfitting. In the Colab notebook, however, both `RandomizedSearchCV` and `GridSearchCV` are applied to the training data only, with cross-validation internally simulating a validation process. So while an explicit validation set is not set aside, the process is still sound and does not leak test set information — provided the final evaluation is done on the untouched test set.
My recommendation is to use the **`train_test_split`** function from **`sklearn.model_selection`** which is better. And use `RandomizedSearchCV` and `GridSearchCV` to train the data with cross-validation. Thank you.
